addons/website_mail_channel/static/src/snippets/s_channel/options.js

```diff
@@ -8,12 +8,38 @@ var wUtils = require('website.utils');
 var _t = core._t;
 
 options.registry.Channel = options.Class.extend({
+    publicChannels: [],
+
+    /**
+     * @override
+     */
+    start() {
+        const prom = this._getPublicChannels().then(channels => {
+            this.publicChannels = channels;
+        });
+        return Promise.all([this._super(...arguments), prom]);
+    },
```

Comment:
It's ok to write it like you did but this would be easier to read, could you change?
```suggestion
    async start() {
        await this._super(...arguments);
        this.publicChannels = await this._getPublicChannels();
    },
```